src/feeder.rs

```diff
@@ -0,0 +1,86 @@
+use log::{info, error};
+use std::thread::{self, JoinHandle};
+
+use crossbeam_channel::Sender;
+use redis::{Connection, Client};
+use anyhow::Result;
+
+use crate::entities::Event;
+
+pub fn start_feeders(sendr: &Sender<Event>, host: &str, port: u16, num_feeders: i32) -> Vec<JoinHandle<()>> {
+    let mut threads = Vec::with_capacity(num_feeders as usize);
+
+    for _ in 0..num_feeders {
+        let mut feeder = Feeder::connect(&host, port).unwrap();
+        let sendr_copy = Sender::clone(sendr);
+        threads.push(
+            thread::spawn(move || {
+                if let Err(e) = feeder.listen(&sendr_copy) {
+                    error!("Feeder hit an error!: {}", e);
```

Comment:
Think about a flow that notifies the main thread that a feeder thread was lost or respawn it but show an error.